stl/src/print.cpp

```diff
@@ -273,7 +273,7 @@ extern "C" {
     _Allocated_string _Allocated_str{};
     _Transcode_result _Transcoded_str{};
 
-    while (true) {
+    for (;;) {

```

Comment:
nit - is this just stylistic preference? I've always found `for (;;)` loops to be a bit strange and less clear than the obvious `while(true)`, but I won't insist / trust your intuition on what's idiomatic.

Reply:
It was previously more than stylistic preference, but now it's just for consistency. As I explained in the PR description:

> We traditionally used "for ever" to avoid "conditional expression is constant" warnings. While the compiler doesn't warn about `true` anymore, our usage remains consistent. (I don't feel super strongly about this one though.)

What changed a few years ago is that the compiler specifically got smarter about conditional expressions that are literally and directly `true` and `false`. I think it'll also accept individual compile-time constants, but it still warns about more complicated expressions.

Reply:
I actually like the scariness of `for (;;)` loop.
It says, like, be wary, there must be a break or a return somewhere in the middle.